vendor/src/github.com/docker/distribution/registry/api/v2/names.go

```diff
@@ -18,7 +18,7 @@ const (
 // RepositoryNameComponentRegexp restricts registry path component names to
 // start with at least one letter or number, with following parts able to
 // be separated by one period, dash or underscore.
-var RepositoryNameComponentRegexp = regexp.MustCompile(`[a-z0-9]+(?:[._-][a-z0-9]+)*`)
+var RepositoryNameComponentRegexp = regexp.MustCompile(`[a-z0-9]+(?:[._-]+[a-z0-9]+)*`)
```

Comment:
This new regex can be simplified into `[a-z0-9]+[a-z0-9._-]*`

But the comment on the preceeding line (20) now needs to be adjusted, as it's no longer "one period, dash or underscore".

But this is just commentary on the code itself. I do not know if this change is truly acceptable. There must be a reason only one was allowed previously.